### ISSUE TITLE
Please merge 81569e5fe360b9d3229b560b756c477c42058828 into dev channel

### ISSUE THREAD

**Issue by lambdabaa** [dart-lang/sdk#37840]
**commit(s) to merge**: 81569e5fe360b9d3229b560b756c477c42058828

**merge instructions**: clean merge or patch CL

**reason**: This change fixes an issue where some code completion requests will crash analysis server when a code completion model is provided / configured.

/cc @dgrove @kevmoo @mit-mit @whesse @athomas

**Comment by lambdabaa** [comment]
```
ariaye@ariaye1:~/sdk/sdk$ git cherry-pick 81569e5fe360b9d3229b560b756c477c42058828
[cherry-pick-37798-fix f77fbf0f03] Ignore ErrorToken when locating previous n tokens for completion ranking
 Date: Mon Aug 12 16:20:25 2019 +0000
 1 file changed, 2 insertions(+), 1 deletion(-)

ariaye@ariaye1:~/sdk/sdk$ ./tools/build.py -a x64 -mrelease
ninja -C out/ReleaseX64 all
ninja: Entering directory `out/ReleaseX64'
[1/1] Regenerating ninja files
[2191/2191] STAMP obj/dart2js_bot.stamp
The build took 212.372 seconds

ariaye@ariaye1:~/sdk/sdk$ dart pkg/analysis_server/test/services/completion/dart/completion_ranking_internal_test.dart 
00:00 +0: getCursorToken areCompletionsEquivalent exact match
00:00 +1: getCursorToken areCompletionsEquivalent function call
00:00 +2: getCursorToken areCompletionsEquivalent field name
00:00 +3: getCursorToken areCompletionsEquivalent different name
00:00 +4: getCursorToken areCompletionsEquivalent method invocation
00:00 +5: getCursorToken getCursorToken AstNode
00:00 +6: getCursorToken getCursorToken Token
00:00 +7: getCursorToken getCursorToken null
00:00 +8: isStopToken null
00:00 +9: isStopToken synthetic
00:00 +10: isStopToken punctuation
00:00 +11: isStopToken alphabetic
00:00 +12: isStringLiteral null
00:00 +13: isStringLiteral empty string
00:00 +14: isStringLiteral basic
00:00 +15: isStringLiteral raw
00:00 +16: isLiteral string
00:00 +17: isLiteral numeric
00:00 +18: isLiteral not literal
00:00 +19: isTokenDot dot
00:00 +20: isTokenDot not dot
00:00 +21: isTokenDot synthetic
00:00 +22: getCurrentToken
00:00 +23: constructQuery
00:00 +24: elementNameFromRelevanceTag
00:00 +25: All tests passed!

```

**Comment by athomas** [comment]
Done: 533f94951d90274e6f0fbe76183782cef37b1181
